go-controller/pkg/retry/obj_retry.go

```diff
@@ -551,6 +555,9 @@ func (r *RetryFramework) WatchResourceFiltered(namespaceForFilteredHandler strin
 				})
 			},
 			UpdateFunc: func(old, newer interface{}) {
+				if r.ResourceHandler.FilterOutResource(newer) {
```

Comment:
I know this works for the current intention of `FilterOutResource`, but the general logic I would use here is
```
if r.ResourceHandler.FilterOutResource(old) && r.ResourceHandler.FilterOutResource(newer) {
    return
}
```